### PR TITLE
[RDY] terminal: fixes for 'scrollback' option

### PR DESCRIPTION
**Pull request by yagebu**
In new terminals the global value should be used as the new local value
for the option.

In normal buffers, the value is always kept at '-1'. Setting the
global value is allowed using either :set or :setglobal in any type of
buffer. This allows setting the option in an init.vim. :setlocal in
non-terminal buffers raises an error.

Ref #6337.

### CODE REVIEW

File: test/functional/terminal/scrollback_spec.lua [hunk 1]
```diff
@@ -441,11 +441,18 @@ describe("'scrollback' option", function()
...
+    execute('setlocal scrollback=42')
     feed('<CR>')
     eq('E474:', string.match(eval("v:errmsg"), "E%d*:"))
   end)
 
+  it('remains unchanged on a normal buffer', function()
+    command('new')
+    execute('set scrollback=42')
+    feed('<CR>')
+    eq(-1, curbufmeths.get_option('scrollback'))
```
Comment (justinmk) [inline test/functional/terminal/scrollback_spec.lua]: Need to also test v:errmsg is empty. And open a terminal, see that it inherits the global setting.

Comment (yagebu) [inline test/functional/terminal/scrollback_spec.lua]: Will do. Seems like the `feed('<CR>')` can also be removed or is there some indeterminism to leave it in for?

Comment (justinmk) [inline test/functional/terminal/scrollback_spec.lua]: yeah, it can be removed since we don't expect an error.

File: src/nvim/option.c [hunk 2]
```diff
@@ -4331,6 +4330,10 @@ set_num_option (
   if ((opt_flags & (OPT_LOCAL | OPT_GLOBAL)) == 0)
     *(long *)get_varp_scope(&(options[opt_idx]), OPT_GLOBAL) = *pp;
 
+  if (!curbuf->terminal) {
+    curbuf->b_p_scbk = -1;
```
Comment (justinmk) [inline src/nvim/option.c]: shouldn't `buf_copy_options()` already have taken care of this?

Comment (yagebu) [inline src/nvim/option.c]: When, say, `:set scrollback=99` is called, `set_num_option` will have `pp == &curbuf->b_p_scbk`, i.e., it validates the local value and above here it copies the local to the global value. So we need to reset the value to `-1` after it's been copied to the global value. This change seemed the least invasive to handle this case.

Comment (justinmk) [inline src/nvim/option.c]: Yep. Can't see a better way.

**Comment by yagebu** [comment]
Squashed and rebased. Travis failure is unrelated.

**Comment by justinmk** [comment]
Merged. Thanks for following through on this!

**Pull request closed without merging**